portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java

```diff
@@ -29,21 +36,38 @@
 
 @WebServlet("/comments")
 public class CommentsServlet extends HttpServlet {
-    private final List<String[]> comments = new ArrayList<>();
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Query query = new Query("Comment");
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+
+        List<Comment> comments = new ArrayList<>();
+        for (Entity entity : results.asIterable()) {
+            long id = entity.getKey().getId();
+            String name = (String) entity.getProperty("name");
+            String content = (String) entity.getProperty("content");
+            Comment comment = new Comment(id, name, content);
+            comments.add(comment);
+        }
+
         response.setContentType("application/json");
         String json = new Gson().toJson(comments);
         response.getWriter().println(json);
     }
 
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-            String[] comment = new String[2];
-            comment[0] = "name: " + request.getParameter("fullname");
-            comment[1] = "comment: " + request.getParameter("comment");
-            comments.add(comment);
+            String name = request.getParameter("fullname");
+            String content = request.getParameter("comment");
+
+            Entity commentEntity = new Entity("Comment");
+            commentEntity.setProperty("name", name);
+            commentEntity.setProperty("content", content);
+
+            DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+            datastore.put(commentEntity);
```

Comment:
does this generate an ID?

Reply:
yes, id was automatically generated in my datastore